### PR TITLE
Display customer company when creating an order in the BO

### LINKED ISSUE #22643: BO ORDER CREATE - Display customer's company, if B2B enabled
#### Is your feature request related to a problem?

When you create an order in BO and searching for a customer, there is no other info than his name and email.

#### Describe the solution you'd like

It would be good to display customer's company, if B2B mode is enabled. (In addition to user group - https://github.com/PrestaShop/PrestaShop/issues/15954)

### PR DESCRIPTION
**Pull request by tom-combet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Customer's company wasn't displayed in customer search result when creating an order from the BO with B2B mode enabled.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22643.
| How to test?      | Enable B2B mode, create a customer with a company and search him when creating an order in the BO.
| Possible impacts? | None.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23889)

### CODE REVIEW

**Comment by atomiix** [comment]
@Kaudaj thanks for your contribution! I seems that you have a merge commit, can you rebase your PR to get rid of it please? Also, I just did a quick review and it seems that you introduce a bc break. It's ok if there is no other way to do it but you should update the PR description and set `BC Breaks?` to `yes` and document this bc break ;)

**Comment by tom-combet** [comment]
Hello @atomiix!

Thanks for your feedback. 
I removed the merge commit, my bad sorry.
Could you get me started for the bc break? I'm not really familiar with it, is it because I used legacy class?

File: src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php
Comment (atomiix) [inline src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php:53]: This is a BC break because you're changing a public method signature, if a module or anything outside of the core was using this class, it won't work anymore after this PR is merged. That's why we need to document it :)

**Comment by atomiix** [comment]
@Kaudaj Just saw your updated PR description, it's all good now 👍

**Review by atomiix — DISMISSED** [review]

File: src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php
Comment (tom-combet) [reply src/Adapter/Customer/QueryHandler/SearchCustomersHandler.php:53]: Okay, thanks for the explanation!

**Review by Progi1984 — COMMENTED** [review]
@Kaudaj Could you add some behat tests ?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Afaik it's not a BC breaks because the class is in final state :)
I approve the PR, but if you can add behat test, it could be awesome

**Comment by tom-combet** [comment]
@PierreRambaud Alright, I edited the description.

@Progi1984 @PierreRambaud Could you confirm me that :
- I should place my scenario in `tests\Integration\Behaviour\Features\Scenario\Order\order_from_bo.feature` ?
- I should place my step(s) in `tests\Integration\Behaviour\Features\Context\Domain\OrderFeatureContext.php` ?

Thanks!

**Comment by Progi1984** [comment]
@Kaudaj As the domain is about customer, I will make it in `tests/Integration/Behaviour/Features/Scenario/Customer` directory.

As example, you can use `tests/Integration/Behaviour/Features/Scenario/Shop/search_shops.feature`.

**Comment by kpodemski** [comment]
@Progi1984 is it still needed after you add `AddressFormat` compatibility? I didn't manage to reproduce this issue on `develop`

**Comment by Progi1984** [comment]
@kpodemski It doesn't seem relative to AddressFormat.

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php]:
```suggestion
    public function assertNoCustomersWasFound(string $searchPhrases): void
```

File: tests/Integration/Behaviour/Features/Scenario/Customer/search_customers.feature
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Scenario/Customer/search_customers.feature:11]: Could you add a test when PS_B2B_ENABLE is set to 0

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig
Comment (PierreRambaud) [inline templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig]: Shouldn't be displayed only when b2B is enabled?

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php]:
```suggestion
                    if (Configuration::get('PS_B2B_ENABLE')) {
```
Can be a variable loaded outside loops

Comment (PierreRambaud) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php]: Maybe add a condition to be sure the company/company Name are not loaded when we are not in B2B mode?

**Review by PierreRambaud — DISMISSED** [review]
Really nice PR, thanks a lot, few comments have been added :)

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php
Comment (tom-combet) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php]: Not 100% sure this is what you expect, but I tried!

**Review by Progi1984 — DISMISSED** [review]

File: templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig
Comment (tom-combet) [reply templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig]: Done!

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
            'isB2BEnabled' => (bool) $configuration->get('PS_B2B_ENABLE'),
```

File: templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig
Comment (Progi1984) [inline templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig]:
```suggestion
        {% if isB2BEnabled %}
```

**Review by Progi1984 — COMMENTED** [review]
Just naming

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php:304]: I like the idea, but what happen if I add a company name, then turn off the B2B setting, and try to save again?
Can you also add this kind of test?

Comment (tom-combet) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php:304]: Hmm I don't get it...
The case with B2B disabled is covered, isn't it?

File: tests/Integration/Behaviour/Features/Scenario/Customer/search_customers.feature
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Scenario/Customer/search_customers.feature]:
```suggestion

```

**Review by tom-combet — COMMENTED** [review]
Commited!

**Comment by kpodemski** [comment]
Hello @Kaudaj, and thank you for your contribution. Could you please rebase your PR once again so we can move it to QA? 💪

**Comment by tom-combet** [comment]
> 
> 
> Hello @Kaudaj, and thank you for your contribution. Could you please rebase your PR once again so we can move it to QA? 💪

Done!

**Review by kpodemski — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

File: tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php
Comment (PierreRambaud) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php:304]: regarding your condition.
- I set the B2B mode
- Add companyName and Company
- Then turning off the B2B mode
- It will raise an exception

Comment (tom-combet) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php:304]: Hmm it shouldn't because `SearchCustomers` query unset `company` customer field and there is no `companyName` field in behat B2B disabled test table. 
Did you manage to reproduce it? I'll investigate it when I have some time left too.

Comment (tom-combet) [reply tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php:304]: @PierreRambaud Can't reproduce it on my side. I'm really not sure to understand what you mean.
When you say "turn off B2B mode", do you refer to the back-office one? Because test is not using the actual configuration value, but the value from the tests environment (set in Given statement) as I understand it.

**Comment by khouloudbelguith** [comment]
Hi @Kaudaj,

It is ok :heavy_check_mark: 
**B2B enabled**
1. company filed well displayed
image
2. I can search with the company field => ok
image
3. If a customer selected 
image
3. I checked a customer/guest created From FO
4. I created customer from BO from customer page & Add new order page
5. I tried to delete some customers (they have already some orders)

**B2B disabled**
image
A customer selected
image


Thanks!

**Comment by Progi1984** [comment]
Thanks @Kaudaj & @khouloudbelguith

**Pull request merged**
